### ISSUE TITLE
Connect-PartnerCenter does not support MFA

### ISSUE THREAD

**Issue by RobBiddle** [Microsoft/Partner-Center-PowerShell#5]
I'd love to use this module but I can't because it doesn't support MFA enabled accounts.

`$ Connect-PartnerCenter -Credential $myO365cred -ApplicationId xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
Connect-PartnerCenter : AADSTSxxxxx: Due to a configuration change made by your administrator, or because you moved to a new location, you must use
multi-factor authentication to access 'xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx'.
Trace ID: e53ccd0c-6e3a-4497-96e2-b9f0f1f4f800
Correlation ID: b8853794-021a-421e-a5c4-70493690f85c
Timestamp: 2018-08-30 15:34:14Z
At line:1 char:1`
+ Connect-PartnerCenter -Credential $myO365cred -ApplicationId xxxxxxx...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Connect-PartnerCenter], AdalServiceException
    + FullyQualifiedErrorId : Microsoft.IdentityModel.Clients.ActiveDirectory.AdalServiceException,Microsoft.Store.PartnerCenter.PowerShell.Commands.ConnectPa
   rtnerCenter`

**Comment by RobBiddle** [comment]
Looks like it does support MFA, but only if the -Credential parameter is not supplied.

**Comment by isaiahwilliams** [comment]
Hi @RobBiddle, 

I have just published version [1.5.1810.2](https://www.powershellgallery.com/packages/PartnerCenter/1.5.1810.2). With this update I have added support for app only authentication. You can now connect using the following 

```powershell
$credential = Get-Credential
Connect-PartnerCenter -Credential $credential -ServicePrincipal -TenantId 64a764b0-3cda-4311-88ff-9e00aeb02dec
```

The username portion of the credentials needs to be the application identifier and the password portion needs to be the application secret. Please note that some parts of the Partner Center API require app + user authentication, so this means you will not be able to perform those operations using this new authentication method. 

Using this new authentication method, you will now be able to create a script that can execute in a headless state. Please let me know if you have any concerns or questions.

**Comment by evilguest** [comment]
Hello everyone, I am trying to use this, but it does not work. 
Connect-PartnerCenter completes Ok; (pops up the browser, and asks for the creds), but when I try to call Get-PartnerOffer, I am getting  `Authentication failed. MFA required. (invalid_grant)`. The user account is MFA-enabled; when I try getting to the Billing part of the Partner Center, the 2FA flow is automatically triggered. But for some reason the dialog invoked from `Connect-PartnerCenter` doesn't perform the second-factor verification, so I get what get.

Is there any way to force the Connect-PartnerCenter to ask for the verification?
